jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/actions/TailLogs.kt

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.actions
+
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.ToggleAction
+import com.intellij.openapi.project.DumbAware
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ClosedSendChannelException
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import software.aws.toolkits.jetbrains.services.cloudwatch.logs.LogStreamActor
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.resources.message
+
+class TailLogs(private val channel: Channel<LogStreamActor.Messages>) :
+    ToggleAction(message("cloudwatch.logs.tail"), null, AllIcons.RunConfigurations.Scroll_down),
+    CoroutineScope by ApplicationThreadPoolScope("TailCloudWatchLogs"),
+    DumbAware {
+    private var isSelected = false
+    var logStreamingJob: Job? = null
```

Comment:
```suggestion
    var logStreamingJob: Job? = null
        private set
        @TestOnly get
```

Hide the setter and mark getter as test only, I only see it used in tests. Correct me if I am wrong.

Reply:
It's test only, I was trying to avoid exposing it but there was no other way to observe its effects. Used suggestion.